ChildMindInstitute/mindlogger-admin
# feat(mi): Participants create/edit analytics (M2-6735)

### 📝 Description




🔗 [Jira Ticket M2-6735](https://mindlogger.atlassian.net/browse/M2-6735)

This PR tracks new/updated Mixpanel events related to adding and editing participants.

Also reduces browser console warnings due to an unrelated misspelt `data-testid` prop.

### 🪤 Peer Testing

Please refer to [JIRA ticket](https://mindlogger.atlassian.net/browse/M2-6735) for details around when events are triggered and what event properties should be attached to those events.

#### Prerequisites:

- To test, you'll need to have access to the **MindLogger Analytics Dev** project in Mixpanel; please let @farmerpaul know if you are planning to do peer testing so that you can be granted access.
- Ensure you have these env vars set locally:
  ```
  REACT_APP_ENV=dev
  REACT_APP_MIXPANEL_TOKEN=c1e1e662f03d9f1363768731077ea000
  REACT_APP_MIXPANEL_FORCE_ENABLE=true
  REACT_APP_DEVELOP_BUILD_VERSION=head
  ```

#### Testing:
* For each table row in the [JIRA ticket](https://mindlogger.atlassian.net/browse/M2-6735):
  1. Follow the instructions under the **Trigger** column to trigger the event.
  2. Open the **MindLogger Analytics Dev** project in Mixpanel, and navigate to the [Events tab](https://mixpanel.com/project/3333580/view/3841316/app/events). Refresh if needed.

      **Expected outcome:** The event matching the **New Event Name** column should have been added to the top of the list.

  4. Expand the event and open the **Your Properties** tab:  
      <img width="832" alt="Screenshot 2024-06-18 at 4 32 53 PM" src="https://github.com/ChildMindInstitute/mindlogger-admin/assets/1282772/f49b7db6-2faf-41aa-9f78-a7ecdd56e8ea">

      **Expected outcome:** The custom properties for that event mentioned in the **Details** column should be listed, as well as **Applet ID** as the JIRA ticket says applies to all events.
* Click the link under the **Distinct ID** column for any of these recently logged events.

    **Expected outcome:** Under the User Profile Properties box on the left, there should be a field called **App Build Number** whose value is **head**.
